### PR TITLE
Added GetVersionedSchemaFilesInOrder helper function to cassandra_test_util.go

### PR DESCRIPTION
**Pull request by roselander**
Also, minor cleanups:

- Deduplicated setUp/tearDown code
- Replaced no-op logger with test logger

### CODE REVIEW

File: common/persistence/tests/cassandra_test_util.go [hunk 1]
```diff
@@ -124,3 +118,56 @@ func TearDownCassandraKeyspace(cfg *config.Cassandra) {
 		panic(fmt.Sprintf("unable to drop Cassandra keyspace: %v", err))
 	}
 }
+
+func GetVersionedSchemaFilesInOrder(schemaDir string, logger log.Logger) []string {
```
Comment (paulnpdev) [inline common/persistence/tests/cassandra_test_util.go:122]: I found this name really hard to parse.  Since this is the only getter for schema files, unless you anticipate adding more I'd vote for just GetSchemaFiles() (and assume the caller can read a doc string) or perhaps GetSchemaFilesSorted() if you prefer.

File: common/persistence/tests/cassandra_test.go [hunk 2]
```diff
@@ -25,6 +25,7 @@
 package tests
 
 import (
+	"go.uber.org/zap/zaptest"
```
Comment (paulnpdev) [inline common/persistence/tests/cassandra_test.go:28]: I don't see any existing dependencies on this package (I just grepped my source tree). I'm leary of adding a new third party dependency without a little justification.  Does zaptest add something special to justify depending on a new package?

File: common/persistence/tests/cassandra_test.go [hunk 3]
```diff
@@ -48,103 +49,87 @@ const (
...
-	factory := cassandra.NewFactory(
-		*cfg,
+type cassandraTestData struct {
+	cfg *config.Cassandra
+	factory *cassandra.Factory
+	logger log.Logger
+}
+
+func setUpCassandraTest(t *testing.T) (cassandraTestData, func()) {
+	var td cassandraTestData
```
Comment (paulnpdev) [inline common/persistence/tests/cassandra_test.go:59]: nit: you use the local var "testData" in other places, "td" here ... consistency would be nice.

**Review by paulnpdev — APPROVED** [review]

File: common/persistence/tests/cassandra_test_util.go (hunk 1 above)
Comment (roselander) [reply common/persistence/tests/cassandra_test_util.go:122]: Done, and added docstring.

File: common/persistence/tests/cassandra_test.go (hunk 2 above)
Comment (roselander) [reply common/persistence/tests/cassandra_test.go:28]: I actually added this dependency based on feedback in: https://github.com/temporalio/temporal/pull/2417#discussion_r793136654

Afaict this is a submodule of zap, which we already depend on.

File: common/persistence/tests/cassandra_test.go (hunk 3 above)
Comment (roselander) [reply common/persistence/tests/cassandra_test.go:59]: Done

File: common/persistence/tests/cassandra_test.go (hunk 2 above)
Comment (paulnpdev) [reply common/persistence/tests/cassandra_test.go:28]: aha, hence my grep failing.  Lgtm.

**Review by wxing1292 — APPROVED** [review]
